### PR TITLE
Add nullability annotations to Java generated clients

### LINKED ISSUE #19600: [REQ][JAVA] Add @Nullable/@Nonnull annotations to POJO attributes and setters
I'm using java generator for webclient, and the generated [POJO classes only contain `@Nullable`/`@Nonnull` annotations for the getters](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/pojo.mustache#L214), but not for the attributes themselves or the setters.

When using tools like [NullAway](https://github.com/uber/NullAway), it causes some issues because you cannot easily use the fluent setters when setting a nullable attribute, because the setter is not annotated with `@Nullable`.

I'd like to add the annotations to the attributes and the setters, so that it's easier to enable tools like NullAway on code using the generated clients.

I will submit a PR for this feature request.

### PR DESCRIPTION
**Pull request by nvervelle**
Motivations:
Have generated clients properly annotated for nullability to be able to check code using them with tools like NullAway (to close #19600)

Modifications:
* First commit :
    * Add nullable_var_annotations template to handle nullability annotation on vars
    * Adapt pojo templates to use the nullability template (except microprofile)
    * Adapt tests
* Second commit :
    * Modifications from export_docs_generators.sh

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request (Java: @bbdouglas, @sreeshas, @jfiala, @lukoyanov, @cbornet, @jeff9finger, @karismann, @Zomzog, @lwlee2608, @martin-mfg)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache
Comment (martin-mfg) [inline modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache:79]: Hi @nvervelle, thanks for the PR!

For `feign`, the annotations on the field are currently only generated if the field is not a container. Moving the `{{>nullable_var_annotations}}` to line 75 should fix that. Other than that the PR looks good to me.

**Review by martin-mfg — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache
Comment (nvervelle) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache]: Hi @martin-mfg !

Thanks for the comment. I modified the template for feign and updated the PR accordingly (I edited each commit so that I still have one commit for the manual changes, and one commit for the changes made by the script)

**Review by martin-mfg — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache
Comment (martin-mfg) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache]: I can't merge PRs. We'll have to wait for `@wing328` to merge this.

Thanks for the heads-up in case further input from you is needed. But hopefully the PR can simply be merged without further changes.

Comment (joscha) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache]: ^ @wing328 the mention didn't work. @nvervelle Unfortunately this PR needs a rebase now, sorry :(

Comment (nvervelle) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/pojo.mustache]: @martin-mfg , @joscha , @wing328 : I've rebased the PR.

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/11327217987/job/31599273360?pr=19617

can you please update the samples when you've time?

**Comment by nvervelle** [comment]
> https://github.com/OpenAPITools/openapi-generator/actions/runs/11327217987/job/31599273360?pr=19617
> 
> can you please update the samples when you've time?

I've rerun the scripts and also rebased : it seems ok now

**Comment by wing328** [comment]
thanks. will merge after all tests pass

**Comment by slobodator** [comment]
Hi @nvervelle,
Sorry, jumping too late.

JavaSpring wasn't affected right, just java clients?
It would be nice to have spring requests annotated with `javax.annotation.Nullable` too.

**Comment by nvervelle** [comment]
Hi @slobodator , I only took care of java clients for the various libraries in Java.

**Pull request merged**
